Modules/SubModule_Import/xASL_imp_NII2BIDS_Session.m

```diff
@@ -0,0 +1,71 @@
+function xASL_imp_NII2BIDS_Session(imPar, bidsPar, studyPar, listSessions, nameSubjectSession, bidsLabel, iSession)
+%xASL_imp_NII2BIDS_SubjectSession NII2BIDS conversion for a single sessions.
+%
+% FORMAT: xASL_imp_NII2BIDS_SubjectSession(imPar, bidsPar, studyPar, listSessions, nameSubjectSession, subjectLabel, iSession)
+% 
+% INPUT:
+%   imPar                 - JSON file with structure with import parameter (STRUCT, REQUIRED)
+%   bidsPar               - Output of xASL_imp_Config (STRUCT, REQUIRED)
+%   studyPar              - JSON file with the BIDS parameters relevant for the whole study (STRUCT, REQUIRED)
+%   listSessions          - list of sessions (CELL ARRAY, REQUIRED, e.g.: {'ASL_1'})
+%   nameSubjectSession    - name of the subject (CELL ARRAY, REQUIRED)
+%   subjectLabel          - subject label (CHAR ARRAY, REQUIRED)
+%   iSession              - Session number (INTEGER, REQUIRED)
+%
+% OUTPUT:
+%  n/a
+%                         
+% -----------------------------------------------------------------------------------------------------------------------------------------------------
+% DESCRIPTION: NII2BIDS conversion for a single sessions.
+% 
+% 1. Make a subject directory
+% 2. Iterate over runs
+%
+% -----------------------------------------------------------------------------------------------------------------------------------------------------
+% EXAMPLE:     n/a
+% __________________________________
+% Copyright 2015-2021 ExploreASL
+
+    %% Unpack bidsLabel struct
+    subjectLabel = bidsLabel.subject;
+    visitLabel = bidsLabel.visit;
+
+    %% 1. Make a subject directory
+    if length(listSessions)>1 || ~isempty(visitLabel)
+        if ~isempty(visitLabel)
+            sessionLabel = ['ses-' visitLabel];
+        else
+            sessionLabel = ['ses-' listSessions{iSession}(5:end)];
+        end
+
+        if ~exist(fullfile(imPar.BidsRoot,['sub-' subjectLabel],sessionLabel),'dir')
+            xASL_adm_CreateDir(fullfile(imPar.BidsRoot,['sub-' subjectLabel],sessionLabel));
+            xASL_adm_CreateDir(fullfile(imPar.BidsRoot,['sub-' subjectLabel],sessionLabel,'perf'));
+        end
+        inSessionPath = fullfile(imPar.TempRoot,nameSubjectSession,listSessions{iSession});
+        outSessionPath = fullfile(imPar.BidsRoot,['sub-' subjectLabel],sessionLabel);
+
+        % Need to add the underscore so that it doesn't need to be added automatically and can be skipped for empty session
+        sessionLabel = ['_' sessionLabel];
+    else
+        % Session label is skipped
+        sessionLabel = '';
+
+        % Only one session - no session labeling
+        if ~exist(fullfile(imPar.BidsRoot,['sub-' subjectLabel]),'dir')
```

Comment:
Same as above. But I would always create a folder for sessions, that makes our life easier

Reply:
Definitely agree, I answered above.

Reply:
Now I understand. But disagree. My understanding was that BIDS encourages you **not** to create a `ses` subfolder when there is only a single session. So it makes your life easier, but goes against BIDS philosophy. We have this here, I would keep it.

Reply:
Agree. BIDS philosophy makes it easier for users here. Sadly that comes with a little bit more work for the developers.